src/MetadataUtility/Metadata/TargetInformation.cs

```diff
@@ -64,10 +65,16 @@ public Stream FileStream
         }
 
         /// <summary>
-        /// Gets all known support files.
+        /// Gets all support files for this target.
         /// Support files are log files, configuration files, etc.
         /// </summary>
-        public Dictionary<string, string> KnownSupportFiles { get; } = new Dictionary<string, string>();
+        public Dictionary<string, SupportFile> TargetSupportFiles { get; } = new Dictionary<string, SupportFile>();
+
+        /// <summary>
+        /// Gets list of each identified support file
+        /// Support files are log files, configuration files, etc.
+        /// </summary>
+        public static List<SupportFile> KnownSupportFiles { get; } = new List<SupportFile>();
```

Comment:
ok I'd suggest reverting this back to a dictionary keyed off of the support file type.

I assume you switched this to a list because your search may return multiple log files per target? The hope would be that is not the case and we can discriminate the correct log file per target.

However, if for some reason there are multiple support files of the same type for a single target, then other solutions might be more useful (like bundling the collection into the `SupportFile` class).

Reply:
Actually, scratch my last; why do we have two collections at all? What's the difference between KnownSupportFiles and TargetSupportFiles?

It looks like one is a list and one is a dictionary? If that is the sole difference you can collapse these into the one type and use a dictionary like a list in most cases.

Reply:
Oh, it's static. I see. You mentioned that. Forgive my iterative drama.

Static, in short, is a no-no. If there is any kind of parallelism or async activity that shares this data structure you're going to run into race conditions. I'd suggest doing a two-phase scan. I'll add more comments where I see this used.